packages/smart-wallet/src/smartWallet.js

```diff
@@ -366,6 +428,87 @@ export const prepareSmartWallet = (baggage, shared) => {
             },
           });
         },
+
+        provideBrandToPurses() {
+          const brandToPurses = provideLazy(
+            walletPurses,
+            this.facets.helper,
+            _k => {
+              /** @type {MapStore<Brand, PurseRecord[]>} */
+              const store = makeScalarBigMapStore('purses by brand', {
+                durable: true,
+              });
+              return store;
+            },
+          );
+          return brandToPurses;
+        },
+
+        /**
+         * Provide a purse given a NameHub of issuers and their
+         * brands.
+         *
+         * We current support only one NameHub, agoricNames, and
+         * hence one purse per brand. But we store an array of them
+         * to facilitate a transition decentralized introductions.
+         *
+         * @param {Brand} brand
+         * @param {ERef<NameHub>} known - namehub with brand, issuer branches
+         * @returns {Promise<Purse | undefined>} undefined if brand is not well known
+         */
+        async providePurseForKnownBrand(brand, known = shared.agoricNames) {
+          const brandToPurses = this.facets.helper.provideBrandToPurses();
+
+          if (brandToPurses.has(brand)) {
+            const purses = brandToPurses.get(brand);
+            if (purses.length > 0) {
+              // For now, we have just 1, so use it.
+              return purses[0].purse;
+            }
+          }
+
+          /**
+           * Introduce an issuer to this wallet.
```

Comment:
why "introduce" instead of "add"?

naming aside, this is adding an issuer even if it's already defined. it's also allowing multiple issuers per [brand,petname] composite key.

I see that's unlikely to happen because of how this is being called, but that makes me wonder why this is parameterized. if it were defined below `const issuer =` it could get the values it needs from the closure. you could even IIFE it without naming the function.

please simplify

Reply:
> why "introduce" instead of "add"?

The idea was "introduce" as in: grant access to, not just "add something to a data structure".
That is: in the sense of [Granovetter diagrams](http://erights.org/elib/capability/ode/overview.html).

I was trying to think thru the flow where users decide which issuers to rely on. But I think this usage might be backwards. Alice the App introduces Bob the user/wallet to Carol the issuer. So the user/wallet is not doing the introducing. Also, Bob doesn't have a choice when Alice introduces Carol to him. And this is all about choosing whether to rely on an issuer.

Does `acceptIssuer` seem reasonable?

Reply:
perfectly

Reply:
> if it were defined below `const issuer =` ...

yea, I'm inclined to just inline it there